src/settings/environment.inc

```diff
@@ -102,8 +102,18 @@ else {
   $instance = 'localhost';
 }
 
-if (!empty($env) && !empty($platform) && !empty($instance)) {
+if (!empty($env) && !empty($platform)) {
+  if (empty($instance) && php_sapi_name() === 'cli') {
+    if (function_exists('drupal_set_message')) {
+      drupal_set_message('SETTINGS_INSTANCE is not set and we were not invoked from the CLI.', 'warning');
```

Comment:
Maybe we should use `trigger_error` here?
```suggestion
      trigger_error('SETTINGS_INSTANCE is not set and we were not invoked from the CLI.', E_USER_WARNING);
```
that would save the drupal_set_message check. drupal_set_message probably won't work at this stage because it requires a session and that requires a database which isn't usually ready yet.

Reply:
So, I started out with tigger_error / user_error but both Drupal and d4d refused to process them into the Docker log, Drupal's log or on screen.

So, I replaced it with drupal_set_message which worked absolutely fine for both HTTP and CLI. Also, drupal_set_message doesn't log to the database so that fact we haven't bootstrapped the database should be a major issue.

Reply:
That's unnerving, I wonder where they went then. Drupal definitely uses sessions to store the set messages and it definitely persists the sessions to the database but I suppose if the message is set and read in the same request then it's safe.